### PR TITLE
changefeed: Fix changefeed fails immediately after being created successfully.

### LINKED ISSUE #2113: Create new changefeed with stopped changefeed checkpoint as start-ts sucessfully, but it will fail immediately
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- Stop a changefeed by "cli changefeed pause xxx"
- After 2 minutes, create a new changefeed with the above stopped changefeed's checkpoint as --start-ts, which is succesful.
- After several seconds, check the new created changefeed, it become to be failed status.

2. What did you expect to see?

- The new created changefeed will not become failed because the stopped changefeed's checkpoint should be saved.

3. What did you see instead?
image

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       Release Version: v5.1.0
       Edition: Community
       Git Commit Hash: 4ca5d03fbe2fe1d2106a29a38b4c99be66b36eb1
       Git Branch: heads/refs/tags/v5.1.0
       UTC Build Time: 2021-06-17 07:24:39
       GoVersion: go1.16.4

        ```

    - TiCDC version (execute `cdc version`):

        ```
        [release-version=v5.1.0] [git-hash=a91a10bb5cd52ee22e56217402003a596a1b5f2b] [git-branch=heads/refs/tags/v5.1.0] [utc-build-time="2021-06-17 07:20:55"]
        ```

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve? 
Fix changefeed fails immediately after being created successfully.

Close #2113 

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

1. create three changefeeds(test1,test2,test3) and stop test3 until it blocks the gcSafepoint advance.
```shell
$ curl http://127.0.0.1:8300/api/v1/changefeeds?state=stopped
[
 {
  "id": "test3",
  "state": "stopped",
  "checkpoint-tso": 425790752871415809,
  "checkpoint-time": "2021-06-21 16:07:00.707",
  "error": null
 }

mysql> select * from mysql.tidb where VARIABLE_NAME="tikv_gc_safe_point";
+--------------------+-------------------------+--------------------------------------------------------------+
| VARIABLE_NAME      | VARIABLE_VALUE          | COMMENT                                                      |
+--------------------+-------------------------+--------------------------------------------------------------+
| tikv_gc_safe_point | 20210621-16:07:00 +0800 | All versions after safe point can be accessed. (DO NOT EDIT) |
+--------------------+-------------------------+--------------------------------------------------------------+
1 row in set (0.01 sec)

```
2. create changefeed test13 and let: `--start-ts = ${test3's checkpointTs}`.
```shell
$ ./bin/cdc cli changefeed create --sink-uri="blackhole://" --start-ts 425790752871415809 --changefeed-id="test13"
Create changefeed successfully!
ID: test13
```

3. test13 works well.
```shell
 $ curl http://127.0.0.1:8300/api/v1/changefeeds?state=normal
[
 {
  "id": "test13",
  "state": "normal",
  "checkpoint-tso": 425791107210674177,
  "checkpoint-time": "2021-06-21 16:29:32.404",
  "error": null
 },
```
Code changes

 - None

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -262,10 +260,8 @@ func (c *changefeed) preflightCheck(captures map[model.CaptureID]*model.CaptureI
 			if status == nil {
 				status = &model.ChangeFeedStatus{
 					// the changefeed status is nil when the changefeed is just created.
-					// the txn in start ts is not replicated at that time,
-					// so the checkpoint ts and resolved ts should less than start ts.
-					ResolvedTs:   c.state.Info.StartTs - 1,
-					CheckpointTs: c.state.Info.StartTs - 1,
+					ResolvedTs:   c.state.Info.StartTs,
+					CheckpointTs: c.state.Info.StartTs,
```
Comment (overvenus) [inline cdc/owner/changefeed.go:264]: Does it mean TiCDC excludes the txn at the start-ts? @amyangfei

**Review by overvenus — DISMISSED** [review]
LGTM

File: cdc/owner/changefeed.go (hunk 1 above)
Comment (amyangfei) [reply cdc/owner/changefeed.go:264]: No, txn with `start-ts` is included, in a reentrant way

Comment (amyangfei) [reply cdc/owner/changefeed.go:264]: No, txn with start-ts is included, in a reentrant way

Comment (overvenus) [reply cdc/owner/changefeed.go:264]: Does this PR change TiCDC behavior?

Comment (asddongmen) [reply cdc/owner/changefeed.go:264]: This logic is the same as the old owner.

**Review by overvenus — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**